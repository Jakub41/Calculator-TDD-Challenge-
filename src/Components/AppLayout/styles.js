import styled from "@emotion/styled";

export const ContentLayout = styled.div`
	width: 100px;
	height: 100px;
`
