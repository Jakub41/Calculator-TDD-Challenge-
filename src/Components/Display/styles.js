import styled from "@emotion/styled";

export const DisplayDiv = styled.div`

`;
