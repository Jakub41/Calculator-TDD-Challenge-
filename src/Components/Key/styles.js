import styled from "@emotion/styled";

export const KeyDiv = styled.div`

`;

export const KeyButton = styled.button`

`;

