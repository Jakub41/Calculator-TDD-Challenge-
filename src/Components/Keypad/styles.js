import styled from "@emotion/styled";

export const KeypadDiv = styled.div`

`;

export const NumbersDiv = styled.div`

`;

export const OperatorsDiv = styled.div`

`;
