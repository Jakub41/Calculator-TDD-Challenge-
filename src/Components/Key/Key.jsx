import React from "react";
import PropTypes from "prop-types";
import { KeyDiv, KeyButton } from "./styles";

const Key = ({ keyAction, keyType, keyValue }) => (
	<>
		<KeyDiv>
			<KeyButton>{keyValue}</KeyButton>
		</KeyDiv>
	</>
);

Key.propTypes = {
	keyAction: PropTypes.func.isRequired,
	keyType: PropTypes.string.isRequired,
	keyValue: PropTypes.string.isRequired,
};

export default Key;
