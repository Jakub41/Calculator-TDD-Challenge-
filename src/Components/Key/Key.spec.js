import React from "react";
import { shallow } from "enzyme";
import Key from "./Key";
import { KeyDiv } from "./styles";

describe("Key", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Key keyAction={jest.fn()} keyType={""} keyValue={""} />);
	});

	it('should render correctly', () => expect(wrapper).toMatchSnapshot());

	it("should render a <KeyDiv />", () => {
		expect(wrapper.find(KeyDiv)).toHaveLength(1);
	});

	it("should render the value of keyValue", () => {
		wrapper.setProps({ keyValue: "test" });
		expect(wrapper.text()).toEqual("test");
	});
});
