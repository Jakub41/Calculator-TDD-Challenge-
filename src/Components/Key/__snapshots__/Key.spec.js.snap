// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key should render correctly 1`] = `
<Fragment>
  <Styled(div)>
    <Styled(button) />
  </Styled(div)>
</Fragment>
`;
